pkg/restore_snapshot.go

```diff
@@ -0,0 +1,207 @@
+	cmd.Flags().StringVar(&opt.restoreOptions.SourceHost, "source-hostname", opt.restoreOptions.SourceHost, "Name of the host whose data will be restored")
+	cmd.Flags().StringSliceVar(&opt.restoreOptions.Snapshots, "snapshots", opt.restoreOptions.Snapshots, "Snapshots to restore")
+
+	cmd.Flags().StringVar(&opt.InvokerKind, "invoker-kind", opt.InvokerKind, "Type of the restore invoker")
+	cmd.Flags().StringVar(&opt.InvokerName, "invoker-name", opt.InvokerName, "Name of the restore invoker")
+
+	//Restore flags
+	cmd.Flags().StringVar(&downloadDir, "download-dir", downloadDir, "Directory where Snapshots get downloaded")
+	cmd.Flags().StringVar(&name, "name", name, "Name of the etcd member to be restored")
+	cmd.Flags().StringVar(&initialAdvertisePeerUrl, "initial-advertise-peer-urls", initialAdvertisePeerUrl, "Initial advertise peer urls for etcd cluster")
+	cmd.Flags().StringVar(&initialCluster, "initial-cluster", initialCluster, "Initial cluster configuration")
+	cmd.Flags().StringVar(&initialClusterToken, "initial-cluster-token", initialClusterToken, "Initial cluster token for etcd cluster")
+	cmd.Flags().StringVar(&dataDir, "data-dir", dataDir, "Data directory for restoring etcd cluster")
+
+	return cmd
+}
+
+func (opt *etcdOptions) downloadSnapshot(targetRef api_v1beta1.TargetRef) error {
+	// apply nice, ionice settings from env
+	var err error
+	opt.setupOptions.Nice, err = v1.NiceSettingsFromEnv()
+	if err != nil {
+		return err
+	}
+	opt.setupOptions.IONice, err = v1.IONiceSettingsFromEnv()
+	if err != nil {
+		return err
+	}
+
+	// init restic wrapper
+	resticWrapper, err := restic.NewResticWrapper(opt.setupOptions)
+	if err != nil {
+		return err
+	}
+
+	// Run restore
+	_, err = resticWrapper.RunRestore(opt.restoreOptions, targetRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (opt *etcdOptions) restoreSnapshot() error {
+	// get app binding
+	appBinding, err := opt.catalogClient.AppcatalogV1alpha1().AppBindings(opt.namespace).Get(context.TODO(), opt.appBindingName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	//Get the restore invoker information
+	opt.invoker, err = invoker.ExtractRestoreInvokerInfo(opt.kubeClient, opt.stashClient, opt.InvokerKind, opt.InvokerName, opt.namespace)
+	if err != nil {
+		return err
+	}
+
+	creds, err := opt.getCredential(appBinding)
+	if err != nil {
+		return err
+	}
+	restoreArgs := []string{"snapshot", "restore"}
+	restoreArgs = append(restoreArgs, creds...)
+
+	restoreArgs = append(restoreArgs, downloadDir)
+	restoreArgs = append(restoreArgs, "--name", name)
+	restoreArgs = append(restoreArgs, "--initial-cluster", initialCluster)
+	restoreArgs = append(restoreArgs, "--initial-cluster-token", initialClusterToken)
+	restoreArgs = append(restoreArgs, "--initial-advertise-peer-urls", initialAdvertisePeerUrl)
+	restoreArgs = append(restoreArgs, "--data-dir", dataDir)
+
+	sh := NewSessionWrapper()
+	sh.SetEnv("ETCDCTL_API", "3")
+	sh.ShowCMD = true
```

Comment:
Remove this line. We can't show command line arguments when there are credentials.